src/target/riscv/riscv-013.c

```diff
@@ -2153,6 +2180,20 @@ static int examine(struct target *target)
 		else
 			LOG_TARGET_INFO(target, "Core %d could not be made part of halt group %d.",
 					info->index, target->smp);
+
+		/* TODO: Resume groups with external input triggers look to be very problematic.
+		 * Neither OpenOCD nor GDB is ready for the case when a target suddenly
+		 * resumes (due to the trigger), without an explicit resume request and
+		 * at arbitrary moment in time. */
+
```

Comment:
As already mentioned in the other comment, this is a principal problem of the system (GDB + OpenOCD). Not something that can be fixed by adjusting a piece of code in OpenOCD.

So I am afraid that putting a TODO inside OpenOCD code is not enough.